web-lib-funcs.pl

```diff
@@ -13522,6 +13522,139 @@ sub get_lock_links_dir
+        last if ($err);
+        close($fh);
+        }
+    $port++;
+    }
+my $wspath = "/$module/ws-".$port;
+my $now = time();
+$miniserv{'websockets_'.$wspath} = "host=127.0.0.1 port=$port wspath=/ user=$remote_user time=$now";
+&put_miniserv_config(\%miniserv);
+&unlock_file(&get_miniserv_config_file());
+&reload_miniserv();
+return $port;
+}
+
+# get_miniserv_websocket_url(port, [host], [module])
+# Returns the URL for a websocket
+sub get_miniserv_websocket_url
+{
+my ($port, $host, $module) = @_;
+$module ||= $module_name;
+my $ws_proto = lc($ENV{'HTTPS'}) eq 'on' ? 'wss' : 'ws';
+my %miniserv;
+&get_miniserv_config(\%miniserv);
+my $http_host_conf = &trim($miniserv{'websocket_host'} || $host);
+if ($http_host_conf) {
+	if ($http_host_conf !~ /^wss?:\/\//) {
+		$http_host_conf = "$ws_proto://$http_host_conf";
+		}
+	$http_host_conf =~ s/[\/]+$//g;
+	}
+my $http_host = $http_host_conf || "$ws_proto://$ENV{'HTTP_HOST'}";
+return "$http_host/$module/ws-$port";
+}
+
+# remove_miniserv_websocket(port, [module])
+# Remove old websocket from miniserv.conf
+sub remove_miniserv_websocket
+{
+my ($port, $module) = @_;
+$module ||= $module_name;
+my %miniserv;
+if ($port) {
+    &lock_file(&get_miniserv_config_file());
+    &get_miniserv_config(\%miniserv);
+    my $wspath = "/$module/ws-".$port;
+    if ($miniserv{'websockets_'.$wspath}) {
+        delete($miniserv{'websockets_'.$wspath});
+        &put_miniserv_config(\%miniserv);
+        &reload_miniserv();
+        }
+    &unlock_file(&get_miniserv_config_file());
+    }
+}
+
+# cleanup_miniserv_websockets([&skip-ports], [module])
+# Called by scheduled status collection to remove any
+# websockets in miniserv.conf that are no longer used
+sub cleanup_miniserv_websockets
+{
+my ($skip, $module) = @_;
+$skip ||= [ ];
+$module ||= $module_name;
+&lock_file(&get_miniserv_config_file());
+my %miniserv;
+&get_miniserv_config(\%miniserv);
+my $now = time();
+my @clean;
+foreach my $k (keys %miniserv) {
+    $k =~ /^websockets_\/$module\/ws-(\d+)$/ || next;
+    my $port = $1;
+    next if (&indexof($port, @$skip) >= 0);
+    my $when = 0;
+    if ($miniserv{$k} =~ /time=(\d+)/) {
+        $when = $1;
+        }
+    if ($now - $when > 60) {
+        # Has been open for a while, check if the port is still in use?
+        my $err;
+        &open_socket("127.0.0.1", $port, my $fh, \$err);
+        if ($err) {
+            # Closed now, can clean up
+            push(@clean, $k);
+            }
+        else {
+            # Still active
+            close($fh);
+            }
+        }
+    }
+if (@clean) {
+    foreach my $k (@clean) {
+        delete($miniserv{$k});
+        }
+    &put_miniserv_config(\%miniserv);
+    &reload_miniserv();
+    }
+&unlock_file(&get_miniserv_config_file());
+}
+
+# get_miniserv_websockets_modules()
+# Returns a list of modules and themes that use websockets
+sub get_miniserv_websockets_modules
+{
+my @rv;
+foreach (&get_all_module_infos(), &list_themes()) {
```

Comment:
Use a proper loop variable instead of $_

Reply:
I had trouble naming this variable nicely, so I decided not to use it at all. Is it really necessary? It's a very simple call.

Reply:
Okay. Fixed now too.

Reply:
Thanks! I just dislike the magical $_ perl variable

Reply:
Sure! No problem

Reply:
Wait, did you actually fix this? I don't see it in the diff..

Reply:
> Wait, did you actually fix this? I don't see it in the diff..

https://github.com/webmin/webmin/pull/2191/files#diff-bbebbbbee3314a0e1686dfb6d99107e442573a1d68b87102c51fe94f3cc18b8dR13649-R13656

Reply:
Thanks! Sometimes I think github's "Files changed" tab is laggy..

Reply:
No problem!